zds/tutorialv2/forms.py

```diff
@@ -197,6 +197,36 @@ def clean(self):
         return cleaned_data
 
 
+class EditContentForm(ContentForm):
+    title = None
+    description = None
+    type = None
+
+    def _create_layout(self):
+        self.helper.layout = Layout(
+            IncludeEasyMDE(),
+            Field("image"),
+            Field("introduction", css_class="md-editor preview-source"),
+            StrictButton(_("Aperçu"), type="preview", name="preview", css_class="btn btn-grey preview-btn"),
+            HTML(
+                '{% if form.introduction.value %}{% include "misc/preview.part.html" \
+                with text=form.introduction.value %}{% endif %}'
+            ),
+            Field("conclusion", css_class="md-editor preview-source"),
+            StrictButton(_("Aperçu"), type="preview", name="preview", css_class="btn btn-grey preview-btn"),
+            HTML(
+                '{% if form.conclusion.value %}{% include "misc/preview.part.html" \
+                with text=form.conclusion.value %}{% endif %}'
+            ),
+            Field("last_hash"),
+            Field("source"),
+            Field("subcategory", template="crispy/checkboxselectmultiple.html"),
+        )
+
+        self.helper.layout.append(Field("msg_commit"))
+        self.helper.layout.append(StrictButton("Valider", type="submit"))
```

Comment:
Pourquoi pas ça :
```suggestion
        Field("msg_commit"),
        StrictButton("Valider", type="submit"),
        )
```
 ? Je vois que c'est pareil dans `ContentForm`, mais puisque c'est du nouveau code, autant faire directement quelque chose de propre :)

Reply:
C'est fait. C'est effectivement mieux de faire comme tu suggères.

Cette série de formulaire a plein d'avertissements dans mon IDE, mais c'est une tâche annexe qui peut être faite séparément ou au fur et à mesure.